### PR TITLE
Added button for MapCreator launch | Some workarounds(#843)

### PR DESCRIPTION
**Pull request by RoiEXLab**
This adresses #843.
Please note, that I removed this do-while loop (why the heck is this even used?!) In order to enable the user to close the map creator once he accidentally opened it without the need to select a file or to kill the process.
This is only a quick workaround...
The future task would be to close the JFileChooser, once the user selects cancel

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/877)

### CODE REVIEW

File: src/games/strategy/engine/framework/startup/ui/EnginePreferences.java [hunk 1]
```diff
@@ -372,6 +377,14 @@ private void setupListeners() {
         ProcessRunnerUtil.exec(commands);
 
     }));
+    m_mapXmlCreator.addActionListener(SwingAction.of("[Beta] Run the Map Creator", e -> {
+      final List<String> commands = new ArrayList<>();
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/startup/ui/EnginePreferences.java]: DRY violation.
Can use a helper method to return a Runnable that takes in 'javaClass' as a parameter

File: src/tools/map/xml/creator/MapXmlUIHelper.java [hunk 2]
```diff
@@ -162,18 +162,16 @@ public static GridBagConstraints getGBCCloneWith(final GridBagConstraints gbcToC
...
-            "No valid selection for " + filename + "!",
-            "Error while selecting " + filename, JOptionPane.ERROR_MESSAGE);
-      } else {
-        return gameXmlFileOpen;
-      }
-    } while (1 > 0);
-
+    final FileOpen gameXmlFileOpen =
+        new FileOpen("Load " + filename, currentDirectory, MapXmlCreator.FILE_NAME_ENDING_XML);
+    if (gameXmlFileOpen.getPathString() != null) {
```
Comment (DanVanAtta) [inline src/tools/map/xml/creator/MapXmlUIHelper.java]: Is this value ever null? If the file chooser window is closed for example - I'm curious what the behavior is in that case.

Comment (RoiEXLab) [inline src/tools/map/xml/creator/MapXmlUIHelper.java]: If the window gets closed the gameXMLFileOpen.getPathString() is null.
In this case A error message is shown, and the window is closed after clicking ok

**Comment by RoiEXLab** [comment]
@DanVanAtta fixed

**Pull request merged**